types/draft-convert/index.d.ts

```diff
@@ -1,73 +1,94 @@
 //                 Munif Tanjim <https://github.com/MunifTanjim>
+//                 Vadim Chistokhvalov <https://github.com/vadim-ch>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.9
+// TypeScript Version: 3.9
 
 // Based on: https://github.com/HubSpot/draft-convert/issues/107#issuecomment-488581709 by <https://github.com/sbusch>
-declare module 'draft-convert' {
-  import {
-    ContentBlock,
-    ContentState,
-    DraftBlockType,
-    DraftInlineStyleType,
-    Entity,
-    RawDraftEntity
-  } from 'draft-js';
-  import { ReactNode } from 'react';
 
-  interface IConvertToHTMLConfig {
-    // Inline styles:
-    styleToHTML?: <T extends DraftInlineStyleType>(style: T) => Tag | void;
+declare module "draft-convert" {
+    import {
+        DraftEntityMutability,
+        RawDraftContentBlock,
+        ContentState,
+        DraftBlockType,
+        DraftInlineStyleType,
+        Entity,
+        RawDraftEntity,
+    } from "draft-js";
+    import { ReactNode } from "react";
 
-    // Block styles:
-    blockToHTML?: (block: ContentBlock) => Tag;
+    type RawDraftContentBlockWithCustomType<T> = Omit<RawDraftContentBlock, "type"> & {
+        type: T;
+    };
 
-    // Entity styling:
-    entityToHTML?: (entity: RawDraftEntity, originalText: string) => Tag;
-  }
+    type ExtendedHTMLElement<T> = (HTMLElement | HTMLLinkElement) & T;
 
-  type EntityKey = string;
+    interface IConvertToHTMLConfig<S, B, E> {
+        // Inline styles:
+        styleToHTML?: (style: S) => Tag | void;
 
-  interface IConvertFromHTMLConfig {
-    // Inline styles:
-    htmlToStyle?: (nodeName: string, node: HTMLElement) => DraftInlineStyleType;
+        // Block styles:
+        blockToHTML?: (block: RawDraftContentBlockWithCustomType<B>) => Tag;
 
-    // Block styles:
-    htmlToBlock?: (
-      nodeName: string,
-      node: HTMLElement
-    ) => DraftBlockType | { type: DraftBlockType; data: object } | false;
+        // Entity styling:
+        entityToHTML?: (entity: E, originalText: string) => Tag;
+    }
 
-    // Html entities
-    htmlToEntity?: (
-      nodeName: string,
-      node: HTMLElement,
-      createEntity: (type: string, mutability: string, data: object) => EntityKey,
-      getEntity: (key: EntityKey) => Entity,
-      mergeEntityData: (key: string, data: object) => void,
-      replaceEntityData: (key: string, data: object) => void
-    ) => void;
+    type EntityKey = string;
 
-    // Text entities
-    textToEntity?: (
-      text: string,
-      createEntity: (type: string, mutability: string, data: object) => EntityKey,
-      getEntity: (key: EntityKey) => Entity,
-      mergeEntityData: (key: string, data: object) => void,
-      replaceEntityData: (key: string, data: object) => void
-    ) => Array<{ entity: EntityKey; offset: number; length: number; result?: string }>;
-  }
+    interface IConvertFromHTMLConfig<S, B, E extends RawDraftEntity> {
```

Comment:
Just as with `IConvertFromHTMLConfig`, providing defaults for type arguments would be preferred.